### PR TITLE
[PL-51080][DOC] Delegate auto-upgrade disabled by default for docker

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Changing the note on [this page](https://developer.harness.io/docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration/) as the customer gets confused due to the existing text.
* Jira/GitHub Issue numbers (if any): https://harness.atlassian.net/browse/PL-51080
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The Harness Delegate for Kubernetes and Helm supports automatic upgrades. Harnes
 Delegate upgrades do not affect pipelines unless the shutdown timeout is reached. Before an upgrade is performed, the delegate finishes the tasks that are underway. The delegate then shuts down. As part of the shutdown process, there is a 10 minute timeout by default. You can configure this setting. For more information, go to [Graceful delegate shutdown](/docs/platform/delegates/delegate-concepts/graceful-delegate-shutdown-process/).
 
 :::info
-The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, it is disabled by default for the Terraform and Docker installation options.
+The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, automatic upgrade is not supported for the Terraform and Docker installation options and is disabled by default.
```
Comment (brian-f-harness) [inline docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md:12]:
```suggestion
The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, the automatic upgrade feature is not supported for the Terraform and Docker installation options and is disabled by default.
```

**Review by brian-f-harness — APPROVED** [review]

File: docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The Harness Delegate for Kubernetes and Helm supports automatic upgrades. Harnes
 Delegate upgrades do not affect pipelines unless the shutdown timeout is reached. Before an upgrade is performed, the delegate finishes the tasks that are underway. The delegate then shuts down. As part of the shutdown process, there is a 10 minute timeout by default. You can configure this setting. For more information, go to [Graceful delegate shutdown](/docs/platform/delegates/delegate-concepts/graceful-delegate-shutdown-process/).
 
 :::info
-The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, it is disabled by default for the Terraform and Docker installation options.
+The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, the automatic upgrade feature is not supported for the Terraform and Docker installation options.
```
Comment (mdmilic) [inline docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md:12]: Hey @saifeemustafaqharness, I think there is some confusion on Terraform. It's supported but disabled by default. We are currently working on enabling it by default. Docker is the one that is just not supported.

**Review by nataraja2018 — APPROVED** [review]

File: docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ The Harness Delegate for Kubernetes and Helm supports automatic upgrades. Harnes
 Delegate upgrades do not affect pipelines unless the shutdown timeout is reached. Before an upgrade is performed, the delegate finishes the tasks that are underway. The delegate then shuts down. As part of the shutdown process, there is a 10 minute timeout by default. You can configure this setting. For more information, go to [Graceful delegate shutdown](/docs/platform/delegates/delegate-concepts/graceful-delegate-shutdown-process/).
 
 :::info
-The automatic upgrade feature is enabled by default for the Kubernetes manifest and Helm installation options. However, it is disabled by default for the Terraform and Docker installation options.
+The automatic upgrade feature is enabled by default for the Kubernetes manifest, Terraform and Helm installation options. However, the automatic upgrade feature is not supported for the Docker installation options.
```
Comment (brian-f-harness) [inline docs/platform/delegates/install-delegates/delegate-upgrades-and-expiration.md:12]:
```suggestion
The automatic upgrade feature is enabled by default for the Kubernetes manifest, Terraform, and Helm installation options. However, the automatic upgrade feature is not supported for the Docker installation option.
```

**Review by brian-f-harness — APPROVED** [review]
Minor suggestions. Otherwise LGTM

**Comment by saifeemustafaqharness** [comment]
@brian-f-harness Can you please approve? I have received Nataraja's approval. Thanks

**Review by nataraja2018 — DISMISSED** [review]

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**